SU2_CFD/src/iteration/CFluidIteration.cpp

```diff
@@ -376,6 +376,8 @@ void CFluidIteration::Solve(COutput* output, CIntegration**** integration, CGeom
   /*--- Boolean to determine if we are running a static or dynamic case ---*/
   bool steady = !config[val_iZone]->GetTime_Domain();
 
+  if (!steady) output->SetCurTimeIter(config[val_iZone]->GetTimeIter());
+
```

Comment:
This is worse, go with what you had initially.

Reply:
Because you are increasing the preconditions necessary to use the output classes correctly. I.e. an additional function must be called.
Moreover, you added changes to one iteration class, but what about the others?
The principle is really simple, minimize the "surface to volume ratio" of abstract classes. Smallest interface possible, and don't duplicate anything, especially logic.

Reply:
Sorry that it's so much work to fix a seemingly simple bug, but most SU2 bugs are because principles like this were  not followed.

Reply:
That's true. Then, I will revert to the previous implementation.

Reply:
> Sorry that it's so much work to fix a seemingly simple bug, but most SU2 bugs are because principles like this were not followed.

No worries, let me know if anything else has to be changed.